web-ui/src/app/pages/multi-step-page/steps.ts

```diff
@@ -40,19 +41,33 @@ interface GlobalStateExampleBased extends GlobalState {
     respectOrder: boolean;
 }
 
+enum StepType {
+    SentenceInput,
+    Matrix,
+    Parse,
+    XpathInput,
+    Analysis,
+    Results,
+    SelectTreebanks
+}
+
 /**
  * A step has a number and a function that performs the necessary actions when entering a step
```

Comment:
Begrijp ik goed dat "necessary actions when entering a step" vooral acties zijn waarbij je moet wachten op een andere service? (of checken of de randvoorwaarden voor de step wel voldaan zijn) Voor mijn begrip, dat ik niet teveel logica op deze plek ga stoppen als ik bv een step toe zou voegen.

Reply:
Ja echt alleen het broodnodige: het is bijvoorbeeld mogelijk om te springen tussen stappen en als dat zou moeten kunnen op basis van de input van de gebruiker, dan moet iedere stap die noodzakelijke informatie aan de staat toevoegen (b.v. een parse van de voorbeeldzin). Maar niet bijvoorbeeld resultaten die kunnen worden afgeleid uit de staat (in een component) én niet hoeven te worden gedeeld.

Reply:
Duidelijk